OpenSim/Simulation/Model/ContactGeometry.cpp

```diff
@@ -133,6 +133,61 @@ void ContactGeometry::updateFromXMLNode(SimTK::Xml::Element& node,
             XMLDocument::addConnector(node, "Connector_PhysicalFrame_",
                     "frame", body_name);
         }
+
+        if (versionNumber < 30507) {
+            // display_preference and color were replaced with Appearance in PR
+            // #1122, at which point the latest XML version number was 30507;
+            // however, the corresponding updateFromXMLNode code below was
+            // added a while later.
+            // https://github.com/opensim-org/opensim-core/pull/1122
+            
+            SimTK::Xml::Element appearanceNode("Appearance");
+
+            // Move color inside Appearance.
+            SimTK::Xml::element_iterator colorIter =
+                        node.element_begin("color");
+            bool addAppearanceNode = false;
+            if (colorIter != node.element_end()) {
+                appearanceNode.insertNodeAfter(appearanceNode.element_end(),
+                        node.removeNode(colorIter));
+                addAppearanceNode = true;
+            } else {
+                // If the user didn't set a color but set one of the other
+                // Appearance properties, then we must set color explicitly so
+                // that the Appearance's default of white is not used.
+                SimTK::Xml::Element color("color");
+                color.setValue("0 1 1"); // SimTK::Cyan
+                appearanceNode.insertNodeAfter(appearanceNode.element_end(),
+                        color);
+            }
+
+            // Move <display_preference> to
+            // <Appearance><SurfaceProperties><representation>
+            SimTK::Xml::element_iterator reprIter =
+                    node.element_begin("display_preference");
+            if (reprIter != node.element_end()) {
+                if (reprIter->getValue() == "0") {
+                    SimTK::Xml::Element visible("visible");
+                    visible.setValue("false");
+                    appearanceNode.insertNodeAfter(appearanceNode.element_end(),
+                            visible);
+                } else {
+                    reprIter->setElementTag("representation");
+                    if (reprIter->getValue() == "4") {
+                        // Enum changed to go 0-3 instead of 0-4
+                        reprIter->setValue("3");
+                    }
+                    SimTK::Xml::Element surfProp("SurfaceProperties");
+                    surfProp.insertNodeAfter(surfProp.element_end(),
+                            node.removeNode(reprIter));
+                    appearanceNode.insertNodeAfter(appearanceNode.element_end(),
+                            surfProp);
+                }
+                addAppearanceNode = true;
```

Comment:
I'm not sure I understand the use of this bool.  AppearanceNode should be added even in the defaults case, or not? In L160 you add the default color, but how will this take effect if the appearanceNode is not added?

Reply:
The default appearance of contact geometry is not simply the default properties of the Appearance class. Therefore, without the bool and the associated logic, if someone altered only the "display_preference" but left the color as default, then the color would not end up as Cyan (which it should). So in L160, I set the color in case we determine later on that the user edited some other properties of the Appearance.

If the user didn't edit any of these properties, there is no need to add the appearance node; we should just use the default property value.

[WrapObject](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Wrap/WrapObject.cpp#L185) has similar logic.

Reply:
Thanks for the clarification @chrisdembia.